pkgs/servers/nextcloud/packages/apps/paths.diff

```diff
@@ -0,0 +1,40 @@
+diff --git a/lib/Settings/SystemConfig.php b/lib/Settings/SystemConfig.php
+index 7814f071..80fc7869 100644
+--- a/lib/Settings/SystemConfig.php
++++ b/lib/Settings/SystemConfig.php
+@@ -124,6 +124,13 @@ class SystemConfig
+      */
+     public static function get(string $key, mixed $default = null): mixed
+     {
++        switch ($key) {
++          case "memories.exiftool": return "@exiftool@";
++          case "memories.vod.ffmpeg": return "@ffmpeg@";
++          case "memories.vod.ffprobe": return "@ffprobe@";
++          case "memories.vod.path": return "@go-vod@";
++        }
++
+         if (!\array_key_exists($key, self::DEFAULTS)) {
+             throw new \InvalidArgumentException("Invalid system config key: {$key}");
+         }
+@@ -154,6 +161,21 @@ public static function get(string $key, mixed $default = null): mixed
+      */
+     public static function set(string $key, mixed $value): void
+     {
++        switch ($key) {
++          case "memories.exiftool":
++            $config->setSystemValue($key, @exiftool@);
++            return;
++          case "memories.vod.ffmpeg":
++            $config->setSystemValue($key, @ffmpeg@);
++            return;
++          case "memories.vod.ffprobe":
++            $config->setSystemValue($key, @ffprobe@);
++            return;
++          case "memories.vod.path":
++            $config->setSystemValue($key, @go-vod@);
++            return;
++        }
++
+         // Check if the key is valid
+         if (!\array_key_exists($key, self::DEFAULTS)) {
+             throw new \InvalidArgumentException("Invalid system config key: {$key}");
```

Comment:
Missing quotes and `Undefined variable $config` at this point, doing it this way does not signal to the end user that their setting change will not be honored.


I would still suggest doing it with the in_array

```suggestion
@@ -154,6 +160,10 @@
      */
     public static function set(string $key, mixed $value): void
     {
+        if ( in_array($key, array( "memories.exiftool", "memories.vod.ffmpeg", "memories.vod.ffprobe", "memories.vod.path")) ){
+            throw new \InvalidArgumentException("Cannot set nix-managed key: {$key}");
+        }
+
         // Check if the key is valid
         if (!\array_key_exists($key, self::DEFAULTS)) {
             throw new \InvalidArgumentException("Invalid system config key: {$key}");

```

Reply:
Not sure if that is a good solution as it will return a 500 for every such request.